dali/operators/image/resize/resize.h

```diff
@@ -35,42 +36,90 @@ namespace detail {
-  std::vector<TransformMeta> per_sample_meta_;
-};
-
 template <typename Backend>
 class Resize : public Operator<Backend>
-             , protected ResizeAttr
-             , protected ResizeBase {
+             , protected ResizeBase<Backend> {
  public:
   explicit Resize(const OpSpec &spec);
 
  protected:
-  bool SetupImpl(std::vector<OutputDesc> &output_desc, const workspace_t<Backend> &ws) override {
-    return false;
+  int NumSpatialDims() const { return resize_attr_.spatial_ndim_; }
+  int FirstSpatialDim() const { return resize_attr_.first_spatial_dim_; }
+
+  bool CanInferOutputs() const override { return true; }
+
+  bool SetupImpl(std::vector<OutputDesc> &output_desc, const workspace_t<Backend> &ws) override;
+
+  void RunImpl(workspace_t<Backend> &ws) override;
+
+  void SaveAttrs(const TensorListView<StorageCPU, int, 1> &shape_data,
+                 const TensorListShape<> &orig_shape) const {
+    int N = orig_shape.num_samples();
+    for (int i = 0; i < N; i++) {
+      auto sample_shape = orig_shape.tensor_shape_span(i);
+      int *out_shape = shape_data.data[i];
+      for (int d = 0; d < NumSpatialDims(); d++) {
+        out_shape[d] = sample_shape[FirstSpatialDim() + d];
+      }
+    }
+  }
+
+  void PrepareParams(const ArgumentWorkspace &ws, const TensorListShape<> &input_shape,
+                     const TensorLayout &layout) {
+    resize_attr_.PrepareResizeParams(spec_, ws, input_shape, layout);
+    assert(NumSpatialDims() >= 1 && NumSpatialDims() <= 3);
+    assert(FirstSpatialDim() >= 0);
+    int N = input_shape.num_samples();
+    resample_params_.resize(N * NumSpatialDims());
+    resampling_attr_.PrepareFilterParams(spec_, ws, N);
+    resampling_attr_.GetResamplingParams(make_span(resample_params_),
+                                         make_cspan(resize_attr_.params_));
   }
 
-  void RunImpl(Workspace<Backend> &ws) override;
-  void SetupSharedSampleParams(Workspace<Backend> &ws) override;
+  void InitializeBackend();
 
   USE_OPERATOR_MEMBERS();
+  std::vector<kernels::ResamplingParams> resample_params_;
+  TensorList<CPUBackend> attr_staging_;
   using Operator<Backend>::RunImpl;
-  bool save_attrs_;
-  int outputs_per_idx_;
+  bool save_attrs_ = false;
+
+  ResizeAttr resize_attr_;
+  ResamplingFilterAttr resampling_attr_;
 };
 
+template <typename Backend>
+bool Resize<Backend>::SetupImpl(std::vector<OutputDesc> &output_desc,
+                                const workspace_t<Backend> &ws) {
+  output_desc.resize(save_attrs_ ? 2 : 1);
+  auto &input = ws.template InputRef<Backend>(0);
+
+  const auto &in_shape = input.shape();
+  auto in_type = input.type().id();
+  auto in_layout = input.GetLayout();
+  int N = in_shape.num_samples();
+
+  PrepareParams(ws, in_shape, in_layout);
+
+  auto out_type = resampling_attr_.GetOutputType(in_type);
+
+  output_desc[0].type = TypeTable::GetTypeInfo(out_type);
+
+  DALI_ENFORCE(ws.NumOutput() == 1 || ws.NumOutput() == 2,
```

Comment:
Does this enforce make any sense. Do the user may trigger this? I think the operator itself defines how many outputs is provides inside the schema itself.